zio-http/jvm/src/test/scala/zio/http/endpoint/openapi/OpenAPIGenSpec.scala

```diff
@@ -223,6 +226,44 @@ object OpenAPIGenSpec extends ZIOSpecDefault {
       .out[SimpleOutputBody]
       .outError[NotFoundError](Status.NotFound)
 
+  private val apiKeyAuthHeaderEndpoint =
+    Endpoint(GET / "withAuthHeader")
+      .in[SimpleInputBody]
+      .header(HttpCodec.Header.apply(apiKeyHeaderName, StringCodec).optional)
+      .out[SimpleOutputBody]
+      .outError[NotFoundError](Status.NotFound)
+
+  private val apiKeyAuthQueryEndpoint =
+    Endpoint(GET / "withAuthQuery")
+      .in[SimpleInputBody]
+      .query(HttpCodec.query[String](apiKeyQueryParamName).optional)
+      .out[SimpleOutputBody]
+      .outError[NotFoundError](Status.NotFound)
+
+  private val basicAuthEndpoint =
+    Endpoint(GET / "withAuthHeader")
+      .in[SimpleInputBody]
+      .header(HttpCodec.authorization)
```

Comment:
The explicit mentioning of the auth header should not be needed. Please remove it. The tests have to be green without it.

Reply:
So we should use `.auth(AuthType)`

Reply:
Yes